server/src/main/java/org/elasticsearch/search/sort/ScriptSortBuilder.java

```diff
@@ -124,6 +128,7 @@ public ScriptSortBuilder(StreamInput in) throws IOException {
         if (in.getVersion().onOrAfter(Version.V_6_1_0)) {
             nestedSort = in.readOptionalWriteable(NestedSortBuilder::new);
         }
```

Comment:
You need to check the version of the input and read the value only if it is on the current version. See above for an example of an option added in v6.1.0.